agent/utils/ephemeral_ports.go

```diff
@@ -195,3 +243,32 @@ func isPortAvailable(port int, protocol string) (bool, error) {
 		return false, errors.New("invalid protocol")
 	}
 }
+
+// PortIsInRange returns true if the given port is within the start-end port range;
+// otherwise, returns false.
+func portIsInRange(port, start, end int) bool {
+	if (port >= start) && (port <= end) {
+		return true
+	}
+	return false
+}
+
+// VerifyPortsWithinRange returns true if the actualPortRange is within the expectedPortRange;
+// otherwise, returns false.
+func verifyPortsWithinRange(actualPortRange, expectedPortRange string) bool {
```

Comment:
probably out-of-scope for this merge-to-dev PR, but is there a case where this would return false?

if the agent is selecting ports always, I am curious why this would ever be false. do we need this extra check?

Reply:
Context - so we had talked about it in the office a bit, we don't think the check would be needed but we also don't think it hurts in the case where something is well and truly off, we wanted to catch that case early.